benchmarks/ALIA/others/array-copyall-bool.sl

```diff
@@ -0,0 +1,21 @@
+(set-logic ALIA)
+
+(synth-inv inv_fun ((a1 (Array Int Bool)) (a2 (Array Int Bool)) (n Int) (i Int)))
+
+(declare-primed-var a1 (Array Int Bool))
+(declare-primed-var a2 (Array Int Bool))
+(declare-primed-var i Int)
+(declare-primed-var n Int)
+
+(define-fun pre_fun ((a1 (Array Int Bool)) (a2 (Array Int Bool)) (n Int) (i Int)) Bool
+  (and (>= n 0) (and (= i 0) (< i n))))
+
+(define-fun trans_fun ((a1 (Array Int Bool)) (a2 (Array Int Bool)) (n Int) (i Int) (a1! (Array Int Bool)) (a2! (Array Int Bool)) (n! Int) (i! Int)) Bool
+  (and (= i! (+ i 1)) (and (= n! n) (and (= a2! a2) (= a1! (store a1 i (select a2 i)))))))
+
+(define-fun post_fun ((a1 (Array Int Bool)) (a2 (Array Int Bool)) (n Int) (i Int)) Bool
+  (or (< i n) (= a1 a2)))
```

Comment:
I don't think this is correct.

If you started with two different arrays `a1` and `a2`, then copied only `0 ... n` indices from `a2` to `a1`, the entire arrays `a1` and `a2` need not be equal at the end.

This could be a fix:

```smt
  (or (< i n)
       (forall ((j Int))
          (=> (and (<= 0 j) (< j n))
              (= (select a1 j)
                 (select a2 j))))))
```